### PR TITLE
Fix unaligned access to FILE_NOTIFY_INFORMATION

### PR DESCRIPTION
**Pull request by ruihe774**
In some environments (e.g. wine), `FILE_NOTIFY_INFORMATION` is packed placed and is unaligned. Reading from unaligned `FILE_NOTIFY_INFORMATION` is a UB and causes panics in debug builds. This PR workarounds this issue by using `ptr::read_unaligned`.

### CODE REVIEW

File: notify/src/windows.rs [hunk 1]
```diff
@@ -331,11 +331,16 @@ unsafe extern "system" fn handle_event(
...
     let mut cur_offset: *const u8 = request.buffer.as_ptr();
-    let mut cur_entry = cur_offset as *const FILE_NOTIFY_INFORMATION;
+    let mut cur_entry = ptr::read_unaligned(cur_offset as *const FILE_NOTIFY_INFORMATION);
     loop {
         // filename length is size in bytes, so / 2
-        let len = (*cur_entry).FileNameLength as usize / 2;
-        let encoded_path: &[u16] = slice::from_raw_parts((*cur_entry).FileName.as_ptr(), len);
+        let len = cur_entry.FileNameLength as usize / 2;
+        let encoded_path: &[u16] = slice::from_raw_parts(
+            // In theory accesing to u16 can also be misaligned, but in reality it is never
```
Comment (dfaust) [inline notify/src/windows.rs:339]: Spelling

**Review by dfaust — CHANGES_REQUESTED** [review]
Thanks for the PR. Looks good to me, but please add some more comments. This is an edge case that probably nobody working on this code will have on their radar.

**Comment by dfaust** [comment]
Thanks. Unfortunately `offset_of` has only been stabilized in Rust 1.77 and our MSRV is 1.72.
My suggestion: I'm going to branch off and release a new version today and then merge this PR into `main` afterwards.

CC: @0xpr03 @JohnTitor

**Comment by ruihe774** [comment]
> Thanks. Unfortunately `offset_of` has only been stabilized in Rust 1.77 and our MSRV is 1.72. My suggestion: I'm going to branch off and release a new version today and then merge this PR into `main` afterwards.
> 
> CC: @0xpr03 @JohnTitor

We can hard code it giving `FILE_NOTIFY_INFORMATION` has a stable ABI; somewhat dirty, though.

**Pull request merged**